bson/bsonrw/extjson_wrappers.go

```diff
@@ -430,17 +430,20 @@ func (ejv *extJSONValue) parseTimestamp() (t, i uint32, err error) {
 
 		switch val.t {
 		case bsontype.Int32:
-			if val.v.(int32) < 0 {
-				return 0, fmt.Errorf("$timestamp %s number should be uint32: %s", key, string(val.v.(int32)))
+			value := val.v.(int32)
+
+			if value < 0 {
+				return 0, fmt.Errorf("$timestamp %s number should be uint32: %d", key, value)
 			}
 
-			return uint32(val.v.(int32)), nil
+			return uint32(value), nil
 		case bsontype.Int64:
-			if val.v.(int64) < 0 || uint32(val.v.(int64)) > math.MaxUint32 {
```

Comment:
The `uint32(val.v.(int64)) > math.MaxUint32` check was causing some numbers larger than `math.MaxUint32` to not be caught. I'm not too sure why, but it's likely because a loss in precision when downcasting the int64 to uint32. Upcasting `math.MaxUint32` to int64 instead fixes it and we know it's in the int64 range so there aren't any precision issues.